modules/tftp/server.rb

```diff
@@ -90,21 +103,107 @@ def pxe_default
+        logger.debug "TFTP: Checking if directory '#{bootloader_path}' exists."
+        unless Dir.exist?(bootloader_path)
+          logger.debug "TFTP: Directory '#{bootloader_path}' does not exist."
+          return
+        end
+      end
+
+      bootloader_path
+    end
+
+    # Configures bootloader files for a host in its host-config directory
+    #
+    # @param mac [String] The MAC address of the host
+    # @param os [String] The lowercase name of the operating system of the host
+    # @param release [String] The major and minor version of the operating system of the host
+    # @param arch [String] The architecture of the operating system of the host
+    # @param bootfilename_efi [String] The architecture specific boot filename suffix
+    def setup_bootloader(mac:, os:, release:, arch:, bootfilename_efi:)
+      pxeconfig_dir_mac = pxeconfig_dir(mac)
+
+      logger.debug "TFTP: Deploying host specific bootloader files to '#{pxeconfig_dir_mac}'."
+
+      FileUtils.mkdir_p(pxeconfig_dir_mac)
+      FileUtils.rm_f(Dir.glob("#{pxeconfig_dir_mac}/*.efi"))
+
+      bootloader_path = bootloader_path(os, release, arch)
+
+      if bootloader_path
+        logger.debug "TFTP: Creating symlinks from bootloader universe."
+
+        Dir.glob(File.join(bootloader_path, '*.efi')).each do |source_file|
+          file_name = File.basename(source_file)
+          symlink_path = File.join(pxeconfig_dir_mac, file_name)
+
+          logger.debug "TFTP: Creating relative symlink: '#{symlink_path}' -> '#{source_file}'"
+          FileUtils.ln_sr(source_file, symlink_path, force: true)
+        end
+      else
+        pxeconfig_dir = pxeconfig_dir()
+        logger.debug "TFTP: Creating symlinks from default bootloader files."
+
+        symlinks = [
+          { source_suffix: 'grub', symlink_suffix: 'boot' },
+          { source_suffix: 'shim', symlink_suffix: 'boot-sb' },
+        ]
+
+        symlinks.each do |link|
+          source_file_path = "#{pxeconfig_dir}/#{link[:source_suffix]}#{bootfilename_efi}.efi"
+          legacy_symlink_path = "#{pxeconfig_dir_mac}/#{link[:source_suffix]}#{bootfilename_efi}.efi"
+          symlink_path = "#{pxeconfig_dir_mac}/#{link[:symlink_suffix]}.efi"
+
+          logger.debug "TFTP: Creating relative symlink: #{legacy_symlink_path} -> #{source_file_path}"
+          FileUtils.ln_sr(source_file_path, legacy_symlink_path, force: true)
+          logger.debug "TFTP: Creating relative symlink: #{symlink_path} -> #{source_file_path}"
+          FileUtils.ln_sr(source_file_path, symlink_path, force: true)
+        end
+      end
+
+      File.write(File.join(pxeconfig_dir_mac, 'os_info'), "#{os} #{release} #{arch}")
+    end
+
+    def del(mac)
+      super mac
+      delete_host_dir mac
+    end
+
+    def pxeconfig_dir(mac = nil)
+      if mac
+        File.join(path, 'host_config', dashed_mac(mac).downcase, 'grub2')
```

Comment:
Did you mean `host-config`?
```suggestion
        File.join(path, 'host-config', dashed_mac(mac).downcase, 'grub2')
```